### PR TITLE
feat: Added missing_recommended_column notice 

### LINKED ISSUE #1350: Create a @RecommendedColumn annotation
## Context
The spec requires certain columns, but having an empty value in them is valid (example [transfers.transfer_type](https://gtfs.org/schedule/reference/#transferstxt)).
The fact that we only had the `@Required` annotation, used in the schema files created [false positives](https://github.com/MobilityData/gtfs-validator/issues/1198).
```
  @Required
  GtfsTransferType transferType();
```
This is why the awesome @briandonahue created a new @RequiredColumn annotation in #1344, designed especially for the case where the column itself is required, but some of the fields can be empty.
```
  @RequiredColumn
  GtfsTransferType transferType();
```

## Proposition
We have a similar situation with `stop_times.timepoint`, except it's not a requirement but rather a recommendation. The GTFS Best Practice mentions (EDIT: they mentioned, it looks like we had another accidental removal in https://github.com/MobilityData/GTFS_Schedule_Best-Practices/pull/44, I am working on it 😅) that the timepoint should be provided, although the values can be empty under certain conditions.

In the validator, we have a [missing_timepoint_column](https://github.com/MobilityData/gtfs-validator/blob/master/RULES.md#missing_timepoint_column) notice that looks for the presence of the column
For consistency with the work done in #1344, we could replace this notice with a `missing_recommended_column` using a `@RecommendedColumn`. 

## Alternatives considered
Keep the validator as-is. I acknowledge this item doesn't add much to the user experience but figured this validator might be easier to maintain if we made this consistent (developers, I would love your POV). 
Additionally, we might want to use the `@RecommendedColumn` for other fields than `stop_times.timepoint` in the future.

### LINKED ISSUE #1455: Replace missing_timepoint_column with missing_recommended_column
After we finish adding a @RecommendedColumn annotation in this validator (issue #1350), we should remove [missing_timepoint_column](https://github.com/MobilityData/gtfs-validator/blob/master/RULES.md#missing_timepoint_column) from the validator and replace by a missing_recommended_column notice using the @RecommendedColumn in `stop_times.timepoint`.

This goes along the lines of the [missing_required_column](https://github.com/MobilityData/gtfs-validator/blob/master/RULES.md#missing_required_column) notice triggered by the @RequiredColumn annotation.

This is more consistent with the rest of the validator and allows us to re-use this notice for other similar use cases that might appear in the future.

### PR DESCRIPTION
**Pull request by jcpitre**
Also removed missing_timepoint_column notice that it replaces.

**Summary:**
Closes #1350 
Closes #1455 

Summarize the changes in the pull request including how it relates to any issues (include the #number, or link them).

In the report, we should not have a missing_recommended_column if the timepoint column if absent from the stop_times.txt fille.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Review by isabelle-dr — APPROVED** [review]
This PR works from a functionality standpoint. I have used one dataset that does not have the `timepoint` column in `stop_times.txt` on the current version and the version in this PR and it works as expected.

Current version: missing_timepoint_column notice.
Screenshot 2023-06-05 at 2 26 32 PM

New version: missing_recommended_column notice.
Screenshot 2023-06-05 at 2 25 12 PM

We will still need a code review for this PR!

File: core/src/test/java/org/mobilitydata/gtfsvalidator/validator/TableHeaderValidatorTest.java [hunk 1]
```diff
@@ -71,13 +75,32 @@ public void missingRequiredColumnShouldGenerateNotice() {
...
+            new CsvHeader(new String[] {"stop_name"}),
+            Set.of("stop_id", "stop_name"),
+            Set.of(),
+            Set.of("stop_id"),
+            container);
+
+    assertThat(container.getValidationNotices())
+        .containsExactly(new MissingRecommendedColumnNotice("stops.txt", "stop_id"));
+    assertThat(container.hasValidationErrors()).isFalse();
+    assertThat(container.hasValidationWarnings()).isTrue();
```
Comment (davidgamez) [inline core/src/test/java/org/mobilitydata/gtfsvalidator/validator/TableHeaderValidatorTest.java:101]: These conditions are already tested by line 99, as the container only holds the MissingRecommendedColumnNotice

Comment (jcpitre) [reply core/src/test/java/org/mobilitydata/gtfsvalidator/validator/TableHeaderValidatorTest.java:101]: Right. I might as well remove something similar in missingRequiredColumnShouldGenerateNotice that I just copied and slightly modified. Line 83.

**Review by davidgamez — APPROVED** [review]
Nice addition!

**Pull request merged**